perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -293,9 +326,145 @@ def _MakeSwiftCommandPrefix(auth_url, tenant_name, username, password):
   return ' '.join(options)
 
 
+def _ParseMultiStreamResults(raw_result, metadata=None):
+  """Read results from the api_multistream worker process.
+
+  Args:
+    raw_result: string. The stdout of the worker process.
+    metadata: dict. Base sample metadata
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+
+  if metadata is None:
+    metadata = {}
+
+  records = json.loads(raw_result)
+  results = []
+
+  for record in records:
+    metric = 'Multi-stream %s KB %s' % (record['size'], record['operation'])
```

Comment:
Please consider include the keyword "latency" as part of the metric name here.

And, I don't think we should include size in the metrics name, it's should be part of the metadata (which shows up as labels in the result), this is because we allow user to specify the size of objects when they run this benchmark, so including the size in the metric name will lead to a proliferation of metric names. Each metric name should represent one scenario (or sub-scenario) that's being measured. In this case, we should have "Multi-stream upload latency", "Multi-stream upload throughput", etc. with number of streams, object sizes as metadata.

One question: we have latency metrics here, will you add throughput metrics as the next PR that does some post-run analysis of aggregated throughputs?